### PR TITLE
Cluster Autoscaler IRSA

### PR DESCRIPTION
**Pull request by vara-bonthu**
*Issue #175 , if available:*

*Description of changes:*
- IRSA for Cluster Autoscaler
- Updates to Node groups IAM policies
- AWS LB variables cleanup
- Updates to examples

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/aws-eks-managed-node-groups/iam.tf [hunk 1]
```diff
@@ -33,16 +33,8 @@ resource "aws_iam_role_policy_attachment" "managed_ng_AmazonEC2ContainerRegistry
...
-  name        = "${var.eks_cluster_id}-${local.managed_node_group["node_group_name"]}-ca"
-  description = "IAM policy for Cluster Autoscaler"
-  path        = var.path
-  policy      = data.aws_iam_policy_document.cluster_autoscaler.json
-}
-
-resource "aws_iam_role_policy_attachment" "cluster_autoscaler" {
-  policy_arn = aws_iam_policy.cluster_autoscaler.arn
+resource "aws_iam_role_policy_attachment" "managed_ng_AmazonSSMManagedInstanceCore" {
+  policy_arn = "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"
```
Comment (askulkarni2) [inline modules/aws-eks-managed-node-groups/iam.tf:37]: Does the worker node actually need this or are we attaching this purely for PVRE purpose? The [template provided in the docs](https://raw.githubusercontent.com/awslabs/amazon-eks-ami/master/amazon-eks-nodegroup.yaml) does not seem to add this.

**Review by kcoleman731 — COMMENTED** [review]
LGTM

File: modules/aws-eks-managed-node-groups/iam.tf (hunk 1 above)
Comment (vara-bonthu) [reply modules/aws-eks-managed-node-groups/iam.tf:37]: This policy is needed for SSMAgent to work on Node Groups(Managed and Self-managed). This is a key requirement for Node roles used for Karpenter as well. Basic template doesn't provide that.

**Review by mbeacom — APPROVED** [review]
lgtm

**Pull request merged**